SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -1118,3 +1139,60 @@ void ASpatialFunctionalTest::ClearAllTakenSnapshots()
 	bWasLoadedFromTakenSnapshot = false;
 	TakenSnapshots.Empty();
 }
+
+void ASpatialFunctionalTest::GenerateMap()
+{
+	GeneratedTestMap->GenerateMap();
+	CreateCustomContentForMap();
+	GeneratedTestMap->AddActorToLevel(GeneratedTestMap->GetWorld()->GetCurrentLevel(), this->GetClass(), FTransform(TestPositionInWorld));
+}
+
+bool ASpatialFunctionalTest::ShouldGenerateMap()
+{
+	return bIsStandaloneTest && this->GetClass() != ASpatialFunctionalTest::StaticClass();
```

Comment:
In `ASpatialFunctionalTest` by default `bIsStandaloneTest` is false. If you want to guard against `ASpatialFunctionalTest` being set to bIsStandaloneTest = true, I would deal that elsewhere.
TL;DR I think `return bIsStandaloneTest` should be enough.